domain/secret/state/state.go

```diff
@@ -587,6 +595,41 @@ WHERE sm.secret_id = $secretID.id
 	return nil
 }
 
+// markObsoleteRevisions obsoletes the revisions and sets the pending_delete to true in the secret_revision table
+// for the specified secret if the revision is not the latest revision and there are no consumers for the revision.
+func markObsoleteRevisions(ctx context.Context, p domain.Preparer, tx *sqlair.TX, uri *coresecrets.URI) error {
```

Comment:
Why is this code moved and no longer a method attached to the st receiver?

Reply:
Since there is no need to be a st's method, also the function has been improved.

Reply:
The method still need the st receiver. We are passing in st as "p" parameter.